File: pig_latin_converter.py
# created by jenny trac
# created on nov 29 2017
# program converts sentences to pig latin

import ui

def convert_word_to_pig_latin(english_word):
    # function converts a single word to pig latin
    
    word_length = len(english_word)
    first_letter_of_word = english_word[0]
    pig_latin_word = ''
    for letter in range(1, word_length):
        pig_latin_word = str(pig_latin_word) + str(english_word[letter])
    pig_latin_word = str(pig_latin_word) + str(first_letter_of_word) + "ay"
    return_statement = str(pig_latin_word)
    
    return return_statement

def convert_touch_up_inside(sender):
    # button to convert entire sentence to pig latin
    
    # variables
    pig_latin_sentence = ' '
    
    # input
    english_sentence = str(view['english_sentence_textfield'].text).split(' ')
    
    # process
    for every_word in english_sentence:
        new_pig_latin_word = convert_word_to_pig_latin(every_word)
        pig_latin_sentence = pig_latin_sentence + " " + str(new_pig_latin_word)
    
    view['converted_label'].text = pig_latin_sentence

view = ui.load_view()
view.present('sheet')

File: PigLatinConverter.java
/* created by Jenny Trac
Created on nov 30 2017
program translates sentences into Pig-latin*/

import java.util.Scanner;

public class PigLatinConverter {
  
    public static String word_converter(String english_word)
    {
    String[] english_word_array = english_word.split("");
    String first_letter = english_word_array[1];
    String new_word = "";    
    for (int counter = 2; counter < english_word_array.length; counter++)
        {
        new_word = new_word + english_word_array[counter];
        }
    new_word = new_word + first_letter + "ay";
    
    return new_word;
    }
    
    public static void main(String[] args)
    {
    // instructions
    System.out.println("Enter a sentence to be translated into Pig Latin:");
    
    // input
    Scanner problem = new Scanner(System.in);
    String[] english_sentence = (problem.nextLine()).split(" ");
    
    //process
    String pig_latin_sentence = "Pig Latin sentence: ";
    for (int count = 0; count < english_sentence.length; count++)
        {
        String pig_latin_word = PigLatinConverter.word_converter(english_sentence[count]);
        pig_latin_sentence = pig_latin_sentence + " " + pig_latin_word;
        }
    
    // output
    System.out.println(pig_latin_sentence);
    }

}
